binding.cc

```diff
@@ -562,16 +572,16 @@ namespace zmq {
 
   NAN_GETTER(Socket::GetPending) {
     Socket* socket = Nan::ObjectWrap::Unwrap<Socket>(info.Holder());
-    info.GetReturnValue().Set(Nan::New<Integer>(socket->pending_));
+    info.GetReturnValue().Set(Nan::New<Boolean>(socket->pending_));
   }
 
   NAN_SETTER(Socket::SetPending) {
-    if (!value->IsNumber()) {
-      Nan::ThrowTypeError("Pending must be an integer");
+    if (!value->IsBoolean()) {
+      Nan::ThrowTypeError("Pending must be a boolean");
```

Comment:
Actually, since the maybe-api changes, it would cause immediate program termination when the following `FromJust` fails.